### PR TITLE
Fixes #687 - Add ability to upload images in comments for an issue

### LINKED ISSUE #687: Add ability to upload image from an issue comment
With #24, we can upload images from new bug reports. We should be able to do so from image comments as well. Since we interact with comments at the JS/API level, this will require extending upload.py to work accodingly (which I think @hallvors will need for some of the tooling he will be working on anyways.

### PR DESCRIPTION
**Pull request by miketaylr**
r? @karlcow for Python bits
r? @magsout for CSS + JS

### CODE REVIEW

**Comment by miketaylr** [comment]
(I still need to fix the flaky tests, the failures are unrelated).

File: webcompat/api/endpoints.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
 from webcompat import cache
 from webcompat import github
 from webcompat import limiter
+from webcompat.helpers import get_comment_data
 from webcompat.helpers import get_headers
 from webcompat.helpers import get_request_headers
 from webcompat.helpers import get_user_info
```
Comment (karlcow) [inline webcompat/api/endpoints.py]: can be scraped. Not used anymore.

File: webcompat/api/endpoints.py [hunk 2]
```diff
@@ -219,10 +220,9 @@ def proxy_comments(number):
...
     if request.method == 'POST':
         try:
-            comment_data = json.loads(request.data)
-            body = json.dumps({"body": comment_data['rawBody']})
             path = 'repos/{0}/{1}/comments'.format(ISSUES_PATH, number)
-            comment = github.raw_request('POST', path, data=body)
+            comment = github.raw_request('POST', path,
+                                         data=get_comment_data(request.data))
             return (json.dumps(comment.json()), comment.status_code,
                     {'content-type': JSON_MIME})
```
Comment (karlcow) [inline webcompat/api/endpoints.py]: Let's see for this full block here.  I was trying to understand what is happening and curious what the function would provide. It simplifies the route :+1: 

But I was wondering why the double dance dict -> json -> dict

Going to http://localhost:5000/issues/276 and posting a text comment with the TEST001 message.

`request.data`

``` python
{"avatarUrl":"https://avatars.githubusercontent.com/u/505230?v=3",
 "body":"<p>TEST001</p>\n",
 "commenter":"karlcow",
 "createdAt":"a few seconds ago",
 "commentLinkId":null,
 "rawBody":"TEST001"}
```

then  `get_comment_data(request.data)` is called.


```python
    comment_data = json.loads(request_data)
```

which is 

```python
{u'body': u'<p>TEST001</p>\n', 
 u'avatarUrl': u'https://avatars.githubusercontent.com/u/505230?v=3', 
 u'rawBody': u'TEST001', 
 u'commenter': u'karlcow', 
 u'commentLinkId': None, 
 u'createdAt': u'a few seconds ago'}
```

(funny how things changed orders)

Then `get_comment_data` returns

```python
 {"body": "TEST001"}
```

to `data` in github.raw_request which is creating a comment object and returns it.

`comment.json()` value is 

```python
{u'body': u'TEST001', 
 u'url': u'https://api.github.com/repos/webcompat/webcompat-tests/issues/comments/139690086', 
 u'created_at': u'2015-09-12T00:01:28Z', 
 u'html_url': u'https://github.com/webcompat/webcompat-tests/issues/276#issuecomment-139690086', 
 u'updated_at': u'2015-09-12T00:01:28Z', 
 u'user': {
    u'following_url': u'https://api.github.com/users/karlcow/following{/other_user}', 
    u'events_url': u'https://api.github.com/users/karlcow/events{/privacy}', 
    u'organizations_url': u'https://api.github.com/users/karlcow/orgs', 
    u'url': u'https://api.github.com/users/karlcow', 
    u'gists_url': u'https://api.github.com/users/karlcow/gists{/gist_id}', 
    u'html_url':u'https://github.com/karlcow', 
    u'subscriptions_url': u'https://api.github.com/users/karlcow/subscriptions', 
    u'avatar_url': u'https://avatars.githubusercontent.com/u/505230?v=3', 
    u'repos_url': u'https://api.github.com/users/karlcow/repos', 
    u'received_events_url': u'https://api.github.com/users/karlcow/received_events', 
    u'gravatar_id': u'', 
    u'starred_url': u'https://api.github.com/users/karlcow/starred{/owner}{/repo}', 
    u'site_admin': False, 
    u'login': u'karlcow', 
    u'type': u'User', 
    u'id': 505230, 
    u'followers_url': u'https://api.github.com/users/karlcow/followers'}, 
u'id': 139690086, 
u'issue_url': 
u'https://api.github.com/repos/webcompat/webcompat-tests/issues/276'}
```

And the dump reserializes it.

**Comment by karlcow** [comment]
@miketaylr r+ for the python part. Tested. Code is working. I have a curiosity question up there, but that's just for curiosity.

**Comment by magsout** [comment]
hum need help I think @miketaylr 

When I try to log in (localhost)
>Something went wrong trying to sign into GitHub. :(

**Comment by karlcow** [comment]
@magsout try to erase session.db first. Then restart

**Comment by magsout** [comment]
@karlcow same error :(

**Comment by miketaylr** [comment]
Bummer @magsout. Two debug requests: 

1) Can you paste the HTTP logs (from the terminal) starting just before you click "Log in"? Linking a gist will be fine.
2) Can you open the network panel of devtools and see if there are any 500 errors or something?

**Comment by magsout** [comment]
@miketaylr 

capture d ecran 2015-09-16 a 07 23 17

**Comment by magsout** [comment]
https://gist.github.com/magsout/9d5c90783c23ae58895c

**Comment by karlcow** [comment]
@magsout you seem to have logged in ?
Did you log out of github too? Not only webcompat.

**Comment by magsout** [comment]
>you seem to have logged in ?

@karlcow yes but no. I'm not logged

>Did you log out of github too? Not only webcompat.

I'm logged on Github. I tried to logged out and logged in...

**Comment by miketaylr** [comment]
Man, webapps are hard. @magsout, I don't recall if I shared the id and secret for the local testing app (that's what will correspond to the part from config.py below) with you or not. I just reset all user tokens there, but if you made your own can you try to "Revoke all users tokens"?

```
# We're running on localhost, use the test application
    GITHUB_CLIENT_ID = os.environ.get('FAKE_ID') or "<secrets!>"
    GITHUB_CLIENT_SECRET = os.environ.get('FAKE_SECRET') or "<secrets!>"
    GITHUB_CALLBACK_URL = "http://localhost:5000/callback"
```

**Comment by miketaylr** [comment]
And after doing that, `rm Session.db` and then try to restart the app and login.

**Comment by magsout** [comment]
>I don't recall if I shared the id and secret for the local testing app 

Heu no, I sent to you my config by email.

> just reset all user tokens there, but if you made your own can you try to "Revoke all users tokens"?

Done, and still same error :(

File: webcompat/static/js/lib/issues.js [hunk 3]
```diff
@@ -95,6 +94,126 @@ issues.TextAreaView = Backbone.View.extend({
   }, 250, {maxWait: 1500})
 });
 
+issues.ImageUploadView = Backbone.View.extend({
+  tagName: 'div',
+  className: 'wc-Form-group',
```
Comment (magsout) [inline webcompat/static/js/lib/issues.js]: Better to add : `js-ImageUploadView`  like backbone

File: webcompat/static/js/lib/issues.js [hunk 4]
```diff
@@ -95,6 +94,126 @@ issues.TextAreaView = Backbone.View.extend({
...
+  _submitButton: $('.js-issue-comment-button'),
+  _loaderImage: $('.js-loader'),
+  template: _.template($('#upload-input-tmpl').html()),
+  render: function() {
+    this.$el.html(this.template()).insertAfter($('textarea'));
+    return this;
+  },
+  inputMap: {
+    'image': {
+      'elm': '.ButtonUpload',
```
Comment (magsout) [inline webcompat/static/js/lib/issues.js]: `js-buttonUpload`

File: webcompat/templates/issue.html [hunk 5]
```diff
@@ -89,11 +89,19 @@ <h1 class="wc-IssueDetail-title wc-Title--l">
...
           <textarea id="Comment-text" class="wc-Comment-wrapper wc-Comment-text" placeholder="Leave a comment"></textarea>
-        <!--</div>-->
+          <script type="text/template" id="upload-input-tmpl">
+          <!-- div class="wc-Form-group" parent created by BackBone -->
+          <form><div class="wc-Form-upload">
+            <span class="wc-Form-upload-icon" aria-hidden="true">
+              <span class="wc-Icon wc-Icon--cloud-upload"></span>
+            </span>
+            <label class="wc-Form-label wc-Form-label--upload" for="image">Attach a screenshot image</label>
+            <input accept=".jpe,.jpg,.jpeg,.gif,.png,.bmp" class="ButtonUpload" id="image" name="image" type="file">
```
Comment (magsout) [inline webcompat/templates/issue.html]: Add `js-buttonUpload`

**Comment by magsout** [comment]
@miketaylr 

Except my comment LGTM r+

**Comment by miketaylr** [comment]
Thanks for the review @magsout! Will make those changes (and spend some time looking at the issue that @karlcow raised).

**Comment by miketaylr** [comment]
> But I was wondering why the double dance dict -> json -> dict

You're awesome @karlcow. :balloon: 

That's just a mistake on my part, I forgot that `raw_request` returns a Requests `Response` object, not a Flask one. So I was just doing extra work for no good reason. Will fix.

**Pull request merged**